### ISSUE TITLE
1.8.9 normal to light brightness insane restriction making more complex models than a (rotated) cube

### ISSUE THREAD

**Issue by LapisSea** [MinecraftForge/MinecraftForge#2439]
So while I was updating to 1.8.9 from 1.8 I noticed that only surfaces with with 0, 90, 180 degrees on all axis (6 sides of a block) will get light calculation and all other will get an "error" brightness that is constant no matter what you do. Of course other angles with brightness are achieved by using Gl11 transformation.
Now I could split all of my models to every quad and transform and render them individually but that is extremely slow rendering.
Here is an example.
http://i.imgur.com/qVg1YOP.png
Note: everything was normal on 1.8 this started in 1.8.9

**Comment by rich1051414** [comment]
I noticed this yesterday but had not looked in to quantifying what was causing it to happen yet. Looks like no need now :+1: 
This will probably be a wide spread problem.

**Comment by LexManos** [comment]
This is something I thought fry fixed with his rendering pipeline update, as getting odd lighting values is an issue in vanilla.
However, how are you rendering it, are you using a normal model and letting MC do the chunk rendering or are you using a TESR to render your things? If so post your related code.

**Comment by LapisSea** [comment]
I am using a TESR since the model is animated. 
tesr src https://goo.gl/0SDWf6

Here is a little summary of what this code is doing. 
Init of the model:
I make all the vertex positions. Than I add them to a rendering buffer/helper and a runnable for before and after rendering
I add that buffer to a class that copies the buffer 6 times and transforms it for all 6 sides of a block so no transformation is needed when rendering. than it calculates all normals for every quad.
Rendering:
the class that has the transformed buffers takes in an array of booleans for each sides and calls the respective transformed buffer.

In short I add the vertices to the rendering buffer and it calculates normals. In the end everything pipes down to this function: https://goo.gl/cuK6sS
Renderer.POS_UV_NORMAL is just a replacement for WorldRender pos,uv,etc.  and Tesselator draw + the format of rendering.

**Comment by williewillus** [comment]
Why are you doing all the transforms yourself? If it's a static model the transform is fast and callrd infrequently. If you use the animation API all caching is done for you

**Comment by LapisSea** [comment]
Oh the transforming is done only if the transformation matrix has effect. It's just a quick way of making multiple models as I sated above. also it is extremely useful to know the positions of transformed vertices in some situations. 
Anyways this is going off topic...

**Comment by RainWarrior** [comment]
Possibly related to #2321. #2437 was just merged, please check if it fixes the problem for you.

**Comment by LapisSea** [comment]
Yep it is working perfectly! :)
http://i.imgur.com/MEPdrnF.png
